lib/authentication/view/profile_page.dart

```diff
@@ -113,20 +80,20 @@ class ProfilePage extends StatelessWidget {
                 User user = snap.data;
                 if (user != null) {
                   userName = user.firstName + ' ' + user.lastName;
-                  userGroup = user.group;
+                  userGroup = user.classes != null && user.classes.isNotEmpty
+                      ? user.classes.last
+                      : null;
```

Comment:
The magic of null-aware operators:
```suggestion
                  userGroup = user?.classes?.last
```
(Important to note that this throws an exception if `classes` is an empty list, but I believe that with the current implementation, it's either `null` or not empty.)